src/test/scala/CoreServerTests.scala

```diff
@@ -0,0 +1,382 @@
+        core.streamMessages(vjh3.id, test_actor_2)
+      }
+
+      "have the stream of messages contain the expected verification result" in {
+        Thread.sleep(20000)
+        assert(actor_tests_results(0) == Some(true))
+        assert(actor_tests_results(1) == Some(true))
+        assert(actor_tests_results(2) == Some(false))
+      }
+
+      "be able to execute 'stop()' without exceptions" in {
+        core.stop()
+      }
+    }
+    "verifying an incorrect viper program several times with caching enabled" should {
+      val config = new ViperConfig(List())
+      config.verify()
+      val core = new ViperCoreServer(config)
+      core.start()
+
+      "produce an OverallFailure Message with a non-empty error list upon first verification." in {
+        val vjh_original = core.verify(verificationError_file, cache_backend, verificationError_ast)
+        val messages_future_original = ViperCoreServerUtils.getMessagesFuture(core, vjh_original.id)
+          while (!messages_future_original.isCompleted) {
+            Thread.sleep(500)
+          }
+          messages_future_original.onComplete({
+            case Success(msgs) =>
+              msgs.last match {
+                case ofm: OverallFailureMessage =>
+                  assert(ofm.result.errors.nonEmpty)
+                case _ => fail()
+              }
+            case Failure(e) => fail()
+          })
+        }
+
+      "produce an OverallFailure Message with an empty error list when re- verified." in {
+        val vjh_cached = core.verify(verificationError_file, cache_backend, verificationError_ast)
+        val messages_future_cached = ViperCoreServerUtils.getMessagesFuture(core, vjh_cached.id)
+        while (!messages_future_cached.isCompleted) {
+          Thread.sleep(100)
+        }
+        messages_future_cached.onComplete({
+          case Success(msgs) =>
+            msgs.last match {
+              case ofm: OverallFailureMessage =>
+                assert(ofm.result.errors.isEmpty)
+              case _ => fail()
+            }
+          case Failure(e) => fail()
+        })
+      }
+
+      "be able to execute 'flushCache()' without exceptions after several verifications" in {
+        core.flushCache()
+      }
+
+      "produce an OverallFailure Message with an empty error list when re- verified after flushing the cache." in {
+        val vjh_flushed = core.verify(verificationError_file, cache_backend, verificationError_ast)
+        val messages_future_flushed = ViperCoreServerUtils.getMessagesFuture(core, vjh_flushed.id)
+        while (!messages_future_flushed.isCompleted) {
+          Thread.sleep(100)
+        }
+        messages_future_flushed.onComplete({
+          case Success(msgs) =>
+            msgs.last match {
+              case ofm: OverallFailureMessage =>
+                assert(ofm.result.errors.nonEmpty)
+              case _ => fail()
+            }
+          case Failure(e) => fail()
+        })
+      }
+
```

Comment:
Maybe here is the place to add the test case that tries verification with enabled caching over a VIper file that has a body-less method.